amqp/connection.py

```diff
@@ -157,6 +158,7 @@ def __init__(self, host='localhost', userid='guest', password='guest',
         requiring certain certificates.
 
         """
+        self.connection_id = uuid.uuid4().hex
```

Comment:
mind making this 'private' ? I don't want ppl to rely on this or make any assumption about it.